client/state/sites/selectors.js

```diff
@@ -102,3 +102,20 @@ export function getSiteSlug( state, siteId ) {
 export function isRequestingSites( state ) {
 	return !! state.sites.fetchingItems.all;
 }
+
+/**
+ * Returns true if an url is already in the current site list
+ * @param {Object}	state Global state tree
+ * @return {Boolean}
+ */
+export function getSiteByUrl( state, url ) {
+	for ( let siteId in state.sites.items ) {
```

Comment:
hmm, state.sites.items is an object, not an array, so (I believe) we would have to use something out of lodash/underscore which internally probably does a `for .. in`. I'm ok if it's more readable, tough. Any suggestion on what to use?